File: invoke-seed.js
import { createClient } from "@supabase/supabase-js";
import dotenv from "dotenv";

dotenv.config();

async function seed() {
    const supabaseUrl = process.env.SUPABASE_URL;
    const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

    if (!supabaseUrl || !supabaseKey) {
        console.error("Missing SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY");
        return;
    }

    try {
        console.log("Invoking seed-knowledge on PROD...");
        const response = await fetch(`${supabaseUrl}/functions/v1/seed-knowledge`, {
            method: 'POST',
            headers: {
                'Authorization': `Bearer ${supabaseKey}`,
                'Content-Type': 'application/json'
            }
        });

        const data = await response.text();
        console.log("Status:", response.status);
        console.log("Response:", data);
    } catch (e) {
        console.error("Error:", e);
    }
}

seed();

File: invoke_seed.py
import urllib.request
import json
import os

supabase_url = os.environ.get('SUPABASE_URL')
supabase_key = os.environ.get('SUPABASE_SERVICE_ROLE_KEY')

if not supabase_url or not supabase_key:
    print("Environment variables missing!")
    exit(1)

url = f"{supabase_url}/functions/v1/seed-knowledge"
headers = {
    'Authorization': f'Bearer {supabase_key}',
    'Content-Type': 'application/json'
}

req = urllib.request.Request(url, headers=headers, method='POST')

try:
    with urllib.request.urlopen(req) as response:
        print("Status:", response.status)
        print("Response:", response.read().decode('utf-8'))
except Exception as e:
    print("Error:", e)
